core-war/src/main/java/org/silverpeas/web/pdc/servlets/PdcSearchRequestRouter.java

```diff
@@ -1028,86 +783,6 @@ private void modifyCriteria(PdcSearchSessionController pdcSC, HttpServletRequest
-    // Create the manager objects
-    containerManager = new ContainerManager();
-    contentManager = new ContentManager();
-
-    // With the global advanced search, we need to know the componentId
-    if (componentId == null) {
-      componentId = pdcSC.getComponentId();
-    }
-    // Get the containerPeas
-    if (bOnlyContainer) {
-      containerPeasPDC = containerManager.getContainerPeasByType("containerPDC");
-    } else {
-      containerPeasPDC = containerManager.getContainerPeas(componentId);
-    }
-
-    // Get the contentPeas
-    if (!bOnlyContainer) {
-      contentPeasPDC = contentManager.getContentPeas(componentId);
-    }
-
-    // Normally we would have to do componentSC.getContainerRoles() and
-    // componentSC.getContentRoles();
-    // Work around (hard coded)
-    // Get the user generic roles
-    // WHEN it would be instanciable !! String[] asUserGenericRoles = componentSC.getUserRoles();
-    // Instead
-    String[] asUserGenericRoles = pdcSC.getUserRoles();
-    List<String> asUserContainerRoles = new ArrayList<>();
-    for (int nI = 0; nI < asUserGenericRoles.length; nI++) {
-      if (asUserGenericRoles[nI].equals("user")) {
-        asUserContainerRoles.add("containerPDC_user");
-      }
-      if (asUserGenericRoles[nI].equals("admin")) {
-        asUserContainerRoles.add("containerPDC_admin");
-      }
-    }
-
-    List<String> asUserContentRoles = new ArrayList<>();
-    if (!bOnlyContainer) {
-      if (contentPeasPDC.getType().equals("fileBoxPlus")
-          || contentPeasPDC.getType().equals("whitePages")
-          || contentPeasPDC.getType().equals("questionReply")) {
-        for (int nI = 0; nI < asUserGenericRoles.length; nI++) {
-          if (asUserGenericRoles[nI].equals("user")) {
-            asUserContentRoles.add("user");
-          }
-          if (asUserGenericRoles[nI].equals("admin")) {
-            asUserContentRoles.add("admin");
-          }
-          if (asUserGenericRoles[nI].equals("publisher")) {
-            asUserContentRoles.add("publisher");
-          }
-          if (asUserGenericRoles[nI].equals("writer")) {
-            asUserContentRoles.add("writer");
-          }
-        }
-      }
-    }
-
-    // Get the content URLIcones
-    List<URLIcone> auContentURLIcones = null;
-    if (!bOnlyContainer) {
-      auContentURLIcones =
-          contentManager.getContentURLIcones(contentPeasPDC.getType(), asUserContentRoles);
-    }
-
-    // Build the Container Workspace
-    ContainerWorkspace containerWorkspace = new ContainerWorkspace();
-    containerWorkspace.setContainerUserRoles(asUserContainerRoles);
-    if (!bOnlyContainer) {
-      containerWorkspace.setContentUserRoles(asUserContentRoles);
-      containerWorkspace.setContentURLIcones(auContentURLIcones);
-    }
-    // Put it in the session controller
-    pdcSC.setContainerWorkspace(containerWorkspace);
-  }
-
   private void buildSearchContext(PdcSearchSessionController pdcSC, HttpServletRequest request)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'buildSearchContext' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUnusedPrivateMethod)